dgea/dgea_scvi.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import pandas as pd
+import scanpy as sc
+import random
+import anndata as ad
+import os
+
+import torch
+from scvi.model import SCANVI, SCVI
+
+def scanvi_dgea(adata:ad.AnnData, groupby:str, reference:str, alternative:str):
+     
+    directory_model = "model_scvi_test/"
+    scanvi_subdirectory = os.path.join(directory_model, "scanvi_model")
+    
+    try:
+        loaded_model = SCVI.load(directory_model, adata=None)
+    except Exception as e:
+        loaded_model = None
+            
+    if isinstance(loaded_model, SCVI):
+        os.makedirs(scanvi_subdirectory, exist_ok=True)
+        scanvi_model = SCANVI.from_scvi_model(loaded_model, unlabeled_category = 'Unknown', labels_key="cell_type")
```

Comment:
This only works because the `cell_type` column exists in the given file. For SIMBA runs where this column exists, there will be a `SCANVI` model. If the cell type annotation does not exist in the SIMBA input, an `SCVI` model will be produced. Thus, if we have an `SCVI` model, the `cell_type` column does not exist.

And AFAIK you also cannot just create the column and fill it with all unknown, because it expects to have at least 2 unique values in the column.

Reply:
A hint on how this will work better:

We can use the existence of the `cell_type` column as an indicator which model type we will have. Based on this, we can perform a conditional import:

```
if <condition>:
    import A as MODEL
else:
    import B as MODEL
```

Subsequently, we can use `MODEL` generally for both model types. The interfaces should be similar enough so that we can use the same methods of the class independently of what type it actually is

Reply:
Aah, ok this would work, yes